andela/ah-shakas
# #162159596 Fix User follow and unfollow

#### What does this PR do?
- Add follow and the following count to the user profile
- Add follow status to the user profile
#### Description of Task to be completed?
Have the following endpoints working:
*  POST /api/profile/{username}/follow
*  DELETE /api/profile/{username}/follow
*  GET /api/profile/{username}/following/
*  GET api/profile/{username}/followers/

#### How should this be manually tested?
* Clone this repository from [here](https://github.com/andela/ah-shakas)
* cd into the repository and run `git checkout bg-fix-user-follow-unfollow-162159596`
* Install requirements by running `pipenv install`
* activate virtual environment by running `pipenv shell`
* run migrations by running command `python manage.py migrate`
* Start the project by running `python manage.py runserver` 
* Using postman,create a user through the endpoint `/api/users/`
* payload for create user is 
              `{
              "user":{
              "username":"jake",
              "email": "jake@jake.jake",
              "password": "jakejake1"
                       }
                }`
* login at endpoint `/api/users/login/` using the same payload as above
* Follow user at endpoint POST `/api/profile/{username}/follow` the username you provide should be 
   of a user who already exists in the system.
* To unfollow a user, do it at endpoint DELETE `/api/profile/{username}/follow`.
* To get the people you are following, go to endpoint GET `/api/profile/{username}/following/`.
* To get your followers, go the endpoint GET `api/profile/{username}/followers/`.
#### Any background context you want to provide?
 * A user should be authenticated before you test.
#### What are the relevant pivotal tracker stories?
- [#162159596](https://www.pivotaltracker.com/n/projects/2198533/stories/162159596)
#### Screenshots (if appropriate)
<img width="891" alt="screen shot 2018-11-26 at 23 40 10" src="https://user-images.githubusercontent.com/28476798/49041006-16f0ad80-f1d5-11e8-95b0-fa6fe2a8f76a.png">
<img width="884" alt="screen shot 2018-11-26 at 23 41 05" src="https://user-images.githubusercontent.com/28476798/49041023-2839ba00-f1d5-11e8-9f3d-ce5ccf35059b.png">
<img width="882" alt="screen shot 2018-11-26 at 23 41 46" src="https://user-images.githubusercontent.com/28476798/49041037-34257c00-f1d5-11e8-9483-0c61e6c796e6.png">

#### Questions:
